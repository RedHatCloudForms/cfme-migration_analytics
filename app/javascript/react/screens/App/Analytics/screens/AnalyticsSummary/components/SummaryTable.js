import React from 'react';
import PropTypes from 'prop-types';
import filesize from 'filesize';

const SummaryRow = ({ name, value }) => (
  <tr>
    <td>{name}</td>
    <td>{value}</td>
  </tr>
);

SummaryRow.propTypes = {
  name: PropTypes.node.isRequired,
  value: PropTypes.node.isRequired
};

const SummaryTable = ({ data }) => (
  <table>
    <tbody>
      {data.numProviders && <SummaryRow name={__('Number of VMware providers')} value={data.numProviders} />}
      <SummaryRow name={__('Number of ESXi hypervisors (2 socket servers) discovered')} value={data.numHypervisors} />
      <SummaryRow name={__('Number of VMs on discovered hypervisors')} value={data.numVms} />
      <SummaryRow
        name={__('Total disk space allocated for discovered VMs')}
        value={filesize(data.allocatedDiskSpace)}
      />
      <SummaryRow
        name={__('Total memory allocated for discovered VMs')}
        value={filesize(data.allocatedMemory * 1024 * 1024)}
      />
      <SummaryRow name={__('Total vCPU cores allocated for discovered VMs')} value={data.numCpuCores} />
    </tbody>
  </table>
);

SummaryTable.propTypes = {
  data: PropTypes.shape({
    numProviders: PropTypes.number,
    numHypervisors: PropTypes.number,
    numVms: PropTypes.number,
    allocatedDiskSpace: PropTypes.number,
    allocatedMemory: PropTypes.number,
    numCpuCores: PropTypes.number
  }).isRequired
};

export default SummaryTable;
