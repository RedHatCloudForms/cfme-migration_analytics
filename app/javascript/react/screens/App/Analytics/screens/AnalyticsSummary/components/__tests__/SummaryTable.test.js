import React from 'react';
import { shallow } from 'enzyme';
import SummaryTable from '../SummaryTable';

describe('summary table', () => {
  const getBaseProps = () => ({
    data: {
      numProviders: 2,
      numHypervisors: 3,
      numVms: 10,
      allocatedDiskSpace: 10737418240,
      allocatedMemory: 16384,
      numCpuCores: 20
    }
  });

  test('renders correctly', () => {
    const component = shallow(<SummaryTable {...getBaseProps()} />);
    expect(component).toMatchSnapshot();
    expect(
      component
        .find('SummaryRow')
        .first()
        .dive()
    ).toMatchSnapshot();
  });
});
