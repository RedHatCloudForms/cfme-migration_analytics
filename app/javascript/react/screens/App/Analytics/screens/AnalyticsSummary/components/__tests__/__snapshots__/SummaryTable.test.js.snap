// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary table renders correctly 1`] = `
<table>
  <tbody>
    <SummaryRow
      name="Number of VMware providers"
      value={2}
    />
    <SummaryRow
      name="Number of ESXi hypervisors (2 socket servers) discovered"
      value={3}
    />
    <SummaryRow
      name="Number of VMs on discovered hypervisors"
      value={10}
    />
    <SummaryRow
      name="Total disk space allocated for discovered VMs"
      value="10 GB"
    />
    <SummaryRow
      name="Total memory allocated for discovered VMs"
      value="16 GB"
    />
    <SummaryRow
      name="Total vCPU cores allocated for discovered VMs"
      value={20}
    />
  </tbody>
</table>
`;

exports[`summary table renders correctly 2`] = `
<tr>
  <td>
    Number of VMware providers
  </td>
  <td>
    2
  </td>
</tr>
`;
